tests/Dfe.PlanTech.Web.UnitTests/Models/ComponentModelTests.cs

```diff
@@ -99,6 +101,38 @@ public void Section_Should_Return_Correct_Maturity()
             Assert.Equal(maturity, lowMaturityRecommendation.Maturity);
         }
 
+        [Fact]
+        public void Section_Should_Return_Correct_Maturity_When_Maturity_Is_A_String()
+        {
+            var maturity = "Low";
+            var section = _componentBuilder.BuildSections().First();
+
+            var lowMaturityRecommendation = section.GetRecommendationForMaturity(maturity);
+
+            Assert.NotNull(lowMaturityRecommendation);
+            Assert.Equal(Maturity.Low, lowMaturityRecommendation.Maturity);
+        }
+
+        [Fact]
+        public void Section_Should_Return_Default_Maturity_When_Maturity_Is_A_String_And_Is_Null()
+        {
+            string? maturity = null;
+            string exceptionText = "Could not find recommendation with maturity Unknown";
+            var section = _componentBuilder.BuildSections().First();
+
+            try
+            {
+                var lowMaturityRecommendation = section.GetRecommendationForMaturity(maturity);
+
+                Assert.NotNull(lowMaturityRecommendation);
+                Assert.Equal(Maturity.Low, lowMaturityRecommendation.Maturity);
+            }
+            catch (KeyNotFoundException ex)
+            {
+                Assert.Equal(exceptionText, ex.Message);
```

Comment:
Would `Assert.Throws` be more applicable?

Reply:
Changed